docs/pages/mdx.mdx

```diff
@@ -0,0 +1,245 @@
+
+## Getting Started
+
+Using MDX in blade is straightforward - simply create a new file with the `.mdx` extension. You can then write markdown and embed React components within your content.
+
+### Basic Example
+
+```mdx
+---
+title: Hello World
+---
+
+# Hello World
+
+Welcome to your first MDX page! You can use **bold text**, *italic text*, and even [links](https://example.com).
+
+<button className="px-4 py-2 bg-blue-500 text-white rounded">
+  Click me
+</button>
+```
+
+### Components
+
+You can use any React component in your MDX files.
+
+```mdx
+import { Button } from '@/components/button';
+
+<Button>Click me</Button>
+```
+
+MDX automatically converts markdown primitives like `#` into HTML elements like `h1`. You can style these with CSS, or enhance them by exporting a `components` object from your `layout.tsx` to customize their rendering.
+
+For example, to add anchor links to all headings:
+
+```tsx
+import { Heading } from '@/components/heading';
+
+export const components = {
+  h1: (props: HeadingProps) => (
+    <Heading
+      level={1}
+      {...props}
+      className="scroll-mt-20"
+    />
+  ),
+  h2: (props: HeadingProps) => (
+    <Heading
+      level={2}
+      {...props}
+      className="scroll-mt-16"
+    />
+  ),
+};
+```
+
+### Table of Contents
+
+For documentation websites, a table of contents helps users navigate your content. Blade automatically generates a table of contents from your MDX headings and passes it to your layout component.
+
+The table of contents structure:
+
+```ts
+type TableOfContents = {
+  value: string;
+  depth: number;
+  children?: TableOfContents[];
+};
+```
+
+You can use this in your layout to create a sidebar navigation:
+
+```tsx
+const DocsLayout = ({
+  children,
+  tableOfContents,
+}: {
+  children: React.ReactNode;
+  tableOfContents: TableOfContents[];
+}) => {
+  return (
+    <div className="flex">
+      <aside className="w-64 bg-white border-r min-h-screen p-4">
+        <nav>
+          {tableOfContents.map((item) => (
+            <a
+              key={item.value}
+              href={`#${item.value.toLowerCase().replace(/\s+/g, '-')}`}
+              className="block py-1 text-sm text-gray-600 hover:text-gray-900"
+            >
+              {item.value}
+            </a>
+          ))}
+        </nav>
+      </aside>
+      
+      <main className="flex-1 p-8">
+        <div className="prose max-w-none">
+          {children}
+        </div>
+      </main>
+    </div>
+  );
+};
+```
+
+### Frontmatter
+
+Frontmatter allows you to add metadata to your MDX files. The `title` field is reserved and automatically sets the page title.
+
+```mdx
+---
+title: Getting Started with Blade
+---
+
+# Getting Started with Blade
+
+Your content here...
+```
+
+This frontmatter is automatically converted to:
+
+```tsx
+import { useMetadata } from '@ronin/blade/server/hooks';
+
+{useMetadata({
+  title: 'Getting Started with Blade'
+})}
+```
```

Comment:
This seems like an implementation detail that shouldn't be shared, since people don't need to think about it. It might also change at any time, since there's no reason for us to use this hook under the hood.